pkg/collectors/pool.go

```diff
@@ -127,6 +136,131 @@ func (f *PerfCollector) initPoolDescs() {
 	}
 }
 
+func isPoolInternalMDisks(poolName string, disksList rest.MDisksList) (bool, error) {
+	var internalPool bool
+	for _, disk := range disksList {
+		if poolName == disk[MdiskGroupNameKey].(string) {
+			if disk[ControllerNameKey].(string) != "" {
+				internalPool = false
+			} else {
+				internalPool = true
+			}
+		}
+	}
+	return internalPool, nil
+}
+
+func isCompressionEnabled(poolName string, disksList rest.MDisksList, fsRestClient *rest.FSRestClient) (bool, error) {
+	var compressionEnabled bool
+	disksInPool := mapMDisksToPool(poolName, disksList)
+
+	for _, diskID := range disksInPool {
+		MDiskInfo, err := fsRestClient.LsSingleMDisk(diskID)
+		if err != nil {
+			log.Errorf("get single mdisk info error: %v", err)
+			return compressionEnabled, err
+		}
+		if MDiskInfo[MdiskEffectiveUsedCapacity].(string) == "" {
+			compressionEnabled = false
+		} else {
+			compressionEnabled = true
+		}
+	}
+	return compressionEnabled, nil
+}
+
+func isPoolArrayMode(poolName string, disksList rest.MDisksList) (bool, error) {
+	var arrayMode bool
+	for _, disk := range disksList {
+		if poolName == disk[MdiskGroupNameKey].(string) {
+			if disk[DiskModekey].(string) != "array" {
+				arrayMode = false
+			} else {
+				arrayMode = true
+			}
+		}
+	}
+	return arrayMode, nil
+}
+
+func (f *PerfCollector) CalcReducedReclaimableCapacityForPool(pool Pool, fsRestClient *rest.FSRestClient, mDisksList rest.MDisksList) (float64, error) {
+	var totalDisksCapacities float64
+	var midSum float64
+	disksInPool := mapMDisksToPool(pool[MdiskNameKey].(string), mDisksList)
+	reclaimable, err := strconv.ParseFloat(pool[ReclaimableKey].(string), 64)
+	if err != nil {
+		log.Errorf("get pool reclaimable capacity failed: %s", err)
+		return -1, err
+	}
+
+	for _, diskID := range disksInPool {
+		PC, EU, physicalFree, err := calcCapacityForSingleDisk(fsRestClient, diskID)
+		if err != nil {
+			log.Errorf("get single disk capacity failed: %s", err)
+			return -1, err
+		}
+		PU := math.Max(0, PC-physicalFree)
+		diskRatio := PC * PU / EU
+
+		totalDisksCapacities += PC
+		midSum += diskRatio
+
+		log.Infof("Calculating reduced reclaimable capacity for Disk ID: %d related to pool %v, "+
+			"PhysicalCapacity PC: %f, MdiskEffectiveUsedCapacity EU: %f, PU: %f, diskRatio: %f, totalDisksCapacities: %f, midSum: %f",
+			diskID, pool[MdiskNameKey].(string), PC, EU, PU, diskRatio, totalDisksCapacities, midSum)
+	}
+
+	if totalDisksCapacities == 0 || midSum == 0 {
+		return 0, nil
+	} else {
+		return reclaimable / totalDisksCapacities * midSum, nil
```

Comment:
maybe we need brackets here?
I am not sure if the formula is
(reclaimable/total) * midSum
OR
reclaimable / (total*midSum)

Reply:
There is no difference ..

Reply:
lets take reclaimable = 10, total = 5, midSum = 3
for the first formula we will get - (10/5) * 3 = 2*3 = 6
for the second formula we will get - 10 / (5*3) = 10 / 15 != 6

Reply:
fixed